lib/core/src/utils.rs

```diff
@@ -86,56 +86,74 @@ pub(crate) async fn derive_fee_rate_sats_per_kvb(
     Ok(result_sat_per_kvb_f32)
 }
 
-pub(crate) fn estimate_refund_fees(
-    swap: &SendSwap,
-    config: &Config,
+pub(crate) async fn new_lbtc_refund_tx(
+    swap_script: LBtcSwapScript,
     output_address: &str,
-    is_cooperative: bool,
-) -> Result<u64, PaymentError> {
-    let swap_script = swap.get_swap_script()?;
-    let electrum_config = ElectrumConfig::new(
-        config.network.into(),
-        &config.liquid_electrum_url,
-        true,
-        true,
-        100,
-    );
-    let swap_tx = boltz_client::LBtcSwapTx::new_refund(
+    config: &Config,
+    liquid_chain_service: Arc<Mutex<dyn LiquidChainService>>,
+) -> Result<LBtcSwapTx> {
+    if swap_script.swap_type == SwapType::ReverseSubmarine {
+        return Err(anyhow!(
+            "Refund Txs cannot be constructed for Reverse Submarine Swaps.".to_string(),
+        ));
+    }
+
+    let output_address = ElementsAddress::from_str(output_address)?;
+
+    let script_pk = swap_script
+        .to_address(config.network.into())
+        .map_err(|_| anyhow!("Could not retrieve address from swap script"))?
+        .to_unconfidential()
+        .script_pubkey();
+
+    let liquid_chain_service = liquid_chain_service.lock().await;
+    let (funding_outpoint, funding_utxo) = liquid_chain_service
+        .get_script_history_outpoint(&script_pk)
+        .await
+        .map_err(|_| anyhow!("Could not retrieve script utxos".to_string()))?;
+
+    let genesis_hash = BlockHash::from_str(match config.network {
+        LiquidNetwork::Mainnet => {
+            "1466275836220db2944ca059a3a10ef6fd2ea684b0688d2c379296888a206003"
+        }
+        LiquidNetwork::Testnet => {
+            "67d5eb1aee63c6c2058a088985503ff0626fd3f7f8022bdc74fab36a359164db"
+        }
+    })
+    .expect("Expecting valid block hash");
```

Comment:
Would avoid `.expect()` as it causes a `panic`. Would throw and propagate an error instead.